diff --git a/node_modules/expo-app-integrity/android/build.gradle b/node_modules/expo-app-integrity/android/build.gradle
index f0d7658..38df657 100644
--- a/node_modules/expo-app-integrity/android/build.gradle
+++ b/node_modules/expo-app-integrity/android/build.gradle
@@ -37,7 +37,7 @@ buildscript {
 
 // Creating sources with comments
 task androidSourcesJar(type: Jar) {
-  classifier = 'sources'
+  archiveClassifier.set("sources")
   from android.sourceSets.main.java.srcDirs
 }
 
@@ -59,15 +59,24 @@ afterEvaluate {
 }
 
 android {
-  compileSdkVersion safeExtGet("compileSdkVersion", 31)
+  compileSdkVersion safeExtGet("compileSdkVersion", 34)
 
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
+    sourceCompatibility JavaVersion.VERSION_17
+    targetCompatibility JavaVersion.VERSION_17
   }
 
   kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_11.majorVersion
+    jvmTarget = JavaVersion.VERSION_17.majorVersion
+  }
+
+  kotlin {
+    jvmToolchain(17)
+  }
+
+  java {
+      sourceCompatibility = JavaVersion.VERSION_17
+      targetCompatibility = JavaVersion.VERSION_17
   }
 
   defaultConfig {
diff --git a/node_modules/expo-app-integrity/ios/IntegrityModule.swift b/node_modules/expo-app-integrity/ios/IntegrityModule.swift
index e3fd69a..039941f 100644
--- a/node_modules/expo-app-integrity/ios/IntegrityModule.swift
+++ b/node_modules/expo-app-integrity/ios/IntegrityModule.swift
@@ -136,11 +136,13 @@ public class IntegrityModule: Module {
             service.isSupported
         }
 
-        AsyncFunction("generateKey") { () async throws -> String in
+        AsyncFunction("generateKey") { (
+            keyIdentifier: String
+        ) async throws -> String in
             
             
             let result = await withCheckedContinuation { continuation in
-                service.generateKey { result, error in
+                service.generateKey { keyIdentifier, error in  /* FIX: was `result' instead of keyIdentifier */
                     
                     return self.appAttestCompletion(
                         result: AppAttestSuccessResult.keyIdentifier(string: result),
