diff --git a/node_modules/expo-file-system/package.json b/node_modules/expo-file-system/package.json
index 9173a21..fda0224 100644
--- a/node_modules/expo-file-system/package.json
+++ b/node_modules/expo-file-system/package.json
@@ -6,6 +6,7 @@
   "types": "build/index.d.ts",
   "sideEffects": false,
   "scripts": {
+    "tsc": "expo-module tsc",
     "build": "expo-module build",
     "clean": "expo-module clean",
     "lint": "expo-module lint",
diff --git a/node_modules/expo-file-system/src/ExponentFileSystem.ts b/node_modules/expo-file-system/src/ExponentFileSystem.ts
index c93fc3c..8444f2c 100644
--- a/node_modules/expo-file-system/src/ExponentFileSystem.ts
+++ b/node_modules/expo-file-system/src/ExponentFileSystem.ts
@@ -1,9 +1,9 @@
-import { NativeModulesProxy } from 'expo-modules-core';
+import { NativeModulesProxy, ProxyNativeModule } from 'expo-modules-core';
 
 import ExponentFileSystemShim from './ExponentFileSystemShim';
 import { ExponentFileSystemModule } from './types';
 
-let platformModule;
+let platformModule: ExponentFileSystemModule|ProxyNativeModule;
 
 if (NativeModulesProxy.ExponentFileSystem) {
   platformModule = NativeModulesProxy.ExponentFileSystem;
@@ -11,4 +11,4 @@ if (NativeModulesProxy.ExponentFileSystem) {
   platformModule = ExponentFileSystemShim;
 }
 
-export default platformModule as ExponentFileSystemModule;
+export default platformModule as ExponentFileSystemModule|ProxyNativeModule;
