diff --git a/node_modules/react-native-send-intent/android/src/main/java/com/burnweb/rnsendintent/RNSendIntentModule.java b/node_modules/react-native-send-intent/android/src/main/java/com/burnweb/rnsendintent/RNSendIntentModule.java
index fcbbcad..1387954 100644
--- a/node_modules/react-native-send-intent/android/src/main/java/com/burnweb/rnsendintent/RNSendIntentModule.java
+++ b/node_modules/react-native-send-intent/android/src/main/java/com/burnweb/rnsendintent/RNSendIntentModule.java
@@ -57,6 +57,8 @@ import okio.BufferedSink;
 import okio.BufferedSource;
 import okio.Okio;
 
+import android.util.Log;
+
 public class RNSendIntentModule extends ReactContextBaseJavaModule {
 
     private static final int FILE_SELECT_CODE = 20190903;
@@ -470,7 +472,16 @@ public class RNSendIntentModule extends ReactContextBaseJavaModule {
 
     @ReactMethod
     public void openApp(String packageName, ReadableMap extras, final Promise promise) {
-        Intent sendIntent = this.reactContext.getPackageManager().getLaunchIntentForPackage(packageName);
+        //Intent sendIntent = this.reactContext.getPackageManager().getLaunchIntentForPackage(packageName);
+        /*
+        Ooops:
+        https://developer.android.com/training/package-visibility
+        getLaunchIntentForPackage() may return null, and returns null in our scenario of pt.lasige.safex.enhmessageme
+        */
+        //Log.d("react-native-send-intent", "packageName: "+packageName);
+        //Log.d("react-native-send-intent", "openApp Intent: "+this.reactContext.getPackageManager().getLaunchIntentForPackage(packageName));
+
+        Intent sendIntent = new Intent("android.intent.action.MAIN");
         if (sendIntent == null) {
             promise.resolve(false);
             return;
@@ -481,6 +492,9 @@ public class RNSendIntentModule extends ReactContextBaseJavaModule {
             return;
         }
 
+        // From the logs: AndroidRuntime: android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?
+        sendIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
         sendIntent.addCategory(Intent.CATEGORY_LAUNCHER);
         this.reactContext.startActivity(sendIntent);
         promise.resolve(true);
@@ -588,7 +602,13 @@ public class RNSendIntentModule extends ReactContextBaseJavaModule {
     @ReactMethod
     public void openAppWithData(String packageName, String dataUri, String mimeType, ReadableMap extras, final Promise promise) {
         Uri uri = Uri.parse(dataUri);
-        Intent sendIntent = new Intent(Intent.ACTION_VIEW);
+
+        Intent sendIntent = new Intent("android.intent.action.VIEW");
+        if (sendIntent == null) {
+            promise.resolve(false);
+            return;
+        }
+        
         if (mimeType != null)
             sendIntent.setDataAndType(uri, mimeType);
         else
@@ -601,7 +621,7 @@ public class RNSendIntentModule extends ReactContextBaseJavaModule {
             return;
         }
 
-        //sendIntent.addCategory(Intent.CATEGORY_LAUNCHER);
+        sendIntent.addCategory(Intent.CATEGORY_DEFAULT);
         sendIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         this.reactContext.startActivity(sendIntent);
         promise.resolve(true);
